clientlibrary/checkpoint/dynamodb-checkpointer.go

```diff
@@ -195,27 +197,36 @@ func (checkpointer *DynamoCheckpoint) GetLease(shard *par.ShardStatus, newAssign
 
 // CheckpointSequence writes a checkpoint at the designated sequence ID
 func (checkpointer *DynamoCheckpoint) CheckpointSequence(shard *par.ShardStatus) error {
```

Comment:
this was previously performing a `PutItem` call which overwrote any existing values. this lead to multiple things consuming from the same shard and caused dynamodb errors when a lease expired since it was writing an empty string for the AssignedTo field.

now it performs an update and overwrites the sequence number and renews the lease too.